### PR TITLE
Reverse order of cached consumption checks

### PR DESCRIPTION
**Pull request by slovdahl**
If a Miele device is temporarily disconnected from the Miele cloud while
running, the Miele API returns a "not connected" state for the device. If the
"not running" check is done first, it causes the energy/water consumption to be
reset to 0. This in turn means that the statistics generated by Home Assistant
will be messed up, it will look like more energy/water was consumed than in
reality.

### CODE REVIEW

**Comment by slovdahl** [comment]
Sorry, this was supposed to be opened against the `dev` branch.

File: custom_components/miele/sensor.py [hunk 1]
```diff
@@ -627,18 +627,25 @@ def state(self):
...
                 or device_state["ecoFeedback"] is None
                 or device_status_value == STATUS_NOT_CONNECTED
             ):
+                # Sometimes the Miele API seems to return a null ecoFeedback
+                # object even though the Miele device is running. Or if the the
+                # Miele device has lost the connection to the Miele cloud, the
+                # status is "not connected". Either way, we need to return the
+                # last known value until the API starts returning something
+                # sane again, otherwise the statistics generated from this
+                # sensor would be messed up.
```
Comment (slovdahl) [inline custom_components/miele/sensor.py:642]: I restored this comment too, I don't know if it was removed by mistake or intentionally @kloknibor? I think it provides some value of why this needs to be done.

**Comment by kloknibor** [comment]
Hi, 

thanks for another contribution. Since you first based it off the master branch a lot of unneeded changes are now in the commit. Is it possible to clean this up a bit? see : https://github.com/HomeAssistant-Mods/home-assistant-miele/pull/90/files Otherwise I just squash it but keeping it clean would be nicer :-) Thanks!

**Comment by slovdahl** [comment]
Yeah, sorry, and no problem, let me just close this and open a new one against the `dev` branch as it should have been.

**Comment by slovdahl** [comment]
#91

**Pull request closed without merging**